src/node_url.cc

```diff
@@ -829,6 +829,16 @@ static url_host_type ParseOpaqueHost(url_host* host,
   return type;
 }
 
+static inline bool IsAllASCII(std::string* input) {
```

Comment:
Actually, it may be even better to just make the checking and lowercasing into the same function (reading 4 bytes at a time as long as possible). That way you're only iterating over the string once.